listenbrainz/webserver/views/art_api.py

```diff
@@ -216,3 +219,91 @@ def cover_art_custom_stats(custom_name, user_name, time_range, image_size):
+        case "Wednesday": most_played_day_message = 'I GOT THROUGH <tspan class="user-stat">WEDNESDAYS</tspan> WITH MUSIC'
+        case "Thursday": most_played_day_message = 'I SPENT TIME WITH MY TUNES ON <tspan class="user-stat">THURSDAYS</tspan>'
+        case "Friday": most_played_day_message = 'I CELEBRATED <tspan class="user-stat">FRIDAYS</tspan> WITH MUSIC'
+        case "Saturday": most_played_day_message = 'I PARTIED HARD - OR HARDLY! - WITH MUSIC ON <tspan class="user-stat">SATURDAYS</tspan>'
+        case "Sunday": most_played_day_message = 'I LOVED SPENDING <tspan class="user-stat">SUNDAYS</tspan> WITH MUSIC'
+        case other: most_played_day_message = f'I CRANKED TUNES ON <tspan class="user-stat">{other}</tspan>'
+
+    most_listened_year = max(stats["most_listened_year"], key=stats["most_listened_year"].get)
+    return render_template(
+        "art/svg-templates/yim-2022.svg",
+        user_name=user_name,
+        most_played_day_message=most_played_day_message,
+        most_listened_year=most_listened_year,
+        total_listen_count=stats["total_listen_count"],
+        total_new_artists_discovered=stats["total_new_artists_discovered"],
+        total_artists_count=stats["total_artists_count"],
+        bg_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-bg.png',
+        magnify_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-magnify.png',
+    )
+
+
+def _cover_art_yim_albums(user_name, stats):
+    """ Create the SVG using YIM top albums for the given year. """
+    cac = CoverArtGenerator(current_app.config["MB_DATABASE_URI"], 3, 250)
+    image_urls = []
+    for item in stats["top_releases"]:
+        if "caa_id" in item and "caa_release_mbid" in item:
+            url = cac.resolve_cover_art(item["caa_id"], item["caa_release_mbid"], 250)
+            image_urls.append(url)
+
+    return render_template(
+        "art/svg-templates/yim-2022-albums.svg",
+        user_name=user_name,
+        image_urls=image_urls,
+        bg_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-bg.png',
+        flames_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-flames.png',
+    )
+
+
+def _cover_art_yim_tracks(user_name, stats):
+    """ Create the SVG using top tracks for the given user. """
+    return render_template(
+        "art/svg-templates/yim-2022-tracks.svg",
+        user_name=user_name,
+        tracks=stats["top_recordings"],
+        bg_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-bg.png',
+        stereo_image_url=f'{current_app.config["SERVER_ROOT_URL"]}/static/img/art/yim-2022-shareable-stereo.png',
+    )
+
+
+
+@art_api_bp.route("/year-in-music/2022/<user_name>", methods=["GET"])
+@crossdomain
+@ratelimit()
+def cover_art_yim_2022(user_name):
```

Comment:
My only concern here is that this endpoint has the year in it, but none of the supporting functions do. I suppose if we change nothing for 2023, then we're fine, but I doubt we'll be doing that. Not urgent right now, but something to think about...

Reply:
I think at the least the theme of the svgs would change next year and we would have to create new templates and supporting functions then anyway.